src/app/store/effects/product.effects.ts

```diff
@@ -0,0 +1,87 @@
+import {Injectable} from '@angular/core';
+import {Actions, createEffect, ofType} from '@ngrx/effects';
+import {
+  addProduct, deleteProduct, loadProducts, loadProductsSuccess, loadProductsFailure, addProductSuccess,
+  addProductFailure, editProduct, editProductSuccess, editProductFailure, deleteProductSuccess, deleteProductFailure,
+  getProduct, getProductSuccess, getProductFailure
+} from '../actions/product.actions';
+import {ProductService} from '../../services/product.service';
+import {of, from, mergeMap} from 'rxjs';
+import {switchMap, map, catchError} from 'rxjs/operators';
+import {Store} from '@ngrx/store';
+import {AppState} from '../state/app.state';
+
+@Injectable()
+export class ProductEffects {
+  constructor(
+    private actions$: Actions,
+    private store: Store<AppState>,
+    private productService: ProductService
+  ) {
+  }
+
+  // Run this code when a load action is dispatched
```

Comment:
Usually some developers prefer not having comments in within the code. The reason is that code should be self-explanatory.
Some short sayings about writing code comments:
- Writing and then maintaining comments is an expense.
- Your compiler doesn’t check your comments so there is no way to determine that comments are correct.